release-notes.txt

```diff
@@ -1,3 +1,565 @@
+RIOT-2021.07 - Release Notes
+============================
+RIOT is a multi-threading operating system which enables soft real-time
+capabilities and comes with support for a range of devices that are typically
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2021.07 release includes:
+
+- Many improvements to ztimer usage
+- Added nucleo-wl55jc and Seeeduino XIAO boards
+- stm32f1 kconfig support
+- overflow fix for malloc thread safety
```

Comment:
Maybe we should throw in the buzz world BadAlloc (see https://msrc-blog.microsoft.com/2021/04/29/badalloc-memory-allocation-vulnerabilities-could-affect-wide-range-of-iot-and-ot-devices-in-industrial-medical-and-enterprise-networks/)

And for ESP, which is not using the wrapper for thread safe allocations, the fix is done separately. Maybe we could just have something along the lines of: "Fixed multiplication overflow in calloc, a.k.a. BadAlloc, for all supported platforms". The implementation details come later on in the list of merged PRs, so IMO we could just go with a superficial one-liner here.

Reply:
Updated and directly amended